test/framework/src/main/java/org/elasticsearch/cluster/routing/RoutingNodesHelper.java

```diff
@@ -14,13 +14,25 @@
 import java.util.Objects;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 import static org.elasticsearch.common.util.CollectionUtils.iterableAsArrayList;
 
 public final class RoutingNodesHelper {
 
     private RoutingNodesHelper() {}
 
+    public static int numberOfShardsOfType(RoutingNodes nodes, ShardRoutingState state) {
```

Comment:
naming nit/suggestion

```suggestion
    public static int numberOfShardsWithState(RoutingNodes nodes, ShardRoutingState state) {
```

Reply:
Good point, I moved it as is, but it is better to rename